### PR TITLE
RockPaperScissors update

### PR DESCRIPTION
**Pull request by soratalee**

### CODE REVIEW

File: RockPaperScissors/RockPaperScissors.cs [hunk 1]
```diff
@@ -6,20 +6,59 @@ class Program
...
+                hand1 = "scissors";
+            }
+
             // Your code here
-            return hand1 + ' ' + hand2;
+            if (hand1 == hand2)
+            {
+                Verdict = "You chose "+hand1+". The Computer chose "+ hand2+"... It's a tie!";
+            }
+            if ((hand1.ToLower() == "rock" && hand2.ToLower() == "scissors") | (hand1.ToLower() == "scissors" && hand2.ToLower() == "paper") | (hand1.ToLower() == "paper" && hand2.ToLower() == "rock"))
```
Comment (daniefer) [inline RockPaperScissors/RockPaperScissors.cs:33]: Good thinking with string comparing, but you already lower cased your input in the Main method.

File: RockPaperScissors/RockPaperScissors.cs [hunk 2]
```diff
@@ -6,20 +6,59 @@ class Program
...
+            {
+                Verdict = "You chose "+hand1+". The Computer chose "+ hand2+"... Computer player wins!";
+            }
+            return Verdict;
+        }
+        public static string ComputerHandGenerator()
+        {
+            String ComputerHand = "";
+            Random r = new Random();
+            int Random = (r.Next(1,4));
```
Comment (daniefer) [inline RockPaperScissors/RockPaperScissors.cs:47]: How could you convert this to an array instead of a series of if statements?

**Review by daniefer — COMMENTED** [review]
Looks like you stopped short of the Spec4 on the project. Make sure to complete the whole assignment https://csharp.austincodingacademy.com/_book/conditionalslesson.html#project

**Comment by daniefer** [comment]
Leaving this assignment as an incomplete (this will show up as a 0 in Campus Manager). Email me when you complete the rest of the assignment and I will regrade for a chance at a higher grade.

**Comment by soratalee** [comment]
Hi Daniel,

can you explain how the code is incomplete?

Is this the code that you were looking into?
using System;
namespace RockPaperScissors
{
class Program
{
public static void Main()
{
string hand1 = "";
bool endGame = false;
//Read user's input
//Try-Catch-Finally for error handling
while (endGame == false)
{
Console.WriteLine("Enter your hand:");
try
{
hand1 = Console.ReadLine().ToLower();
}
catch
{
Console.WriteLine("Error. Please enter a valid hand
('rock','paper','scissors')!");
}
finally
{
Console.WriteLine("Displaying results...");
}
string hand2 = ComputerHandGenerator();
if (hand1 == "bazooka")
{
Console.WriteLine("You have played Bazooka. You win despite the Computer's
hand!");
Console.ReadLine();
//If user enters 'bazooka', the game will end with the user's win condition
being met.
break;
}
//Tests 1 nd 2 for valid input
if (test1(hand1) && test1(hand2))
{
Console.WriteLine(CompareHands(hand1, hand2));
endGame = true;
}
else
{
Console.WriteLine("Error. Please enter a valid hand
('rock','paper','scissors')!");
}
// leave this command at the end so your program does not close
automatically
Console.ReadLine();
}
}
public static string CompareHands(string hand1, string hand2)
{
String Verdict = "";
//Fixes for scissor text
if (hand1.ToLower() == "scissor")
{
hand1 = "scissors";
}
//Win/Loss/Tie logic Returns the outcome to Main
else if (hand1 == hand2)
{
Verdict = "You chose " + hand1 + ". The Computer chose " + hand2 + "...
It's a tie!";
}
else if ((hand1.ToLower() == "rock" && hand2.ToLower() == "scissors") | (
hand1.ToLower() == "scissors" && hand2.ToLower() == "paper") | (hand1.
ToLower() == "paper" && hand2.ToLower() == "rock"))
{
Verdict = "You chose " + hand1 + ". The Computer chose " + hand2 + "... You
win!";
}
else if ((hand1.ToLower() == "scissors" && hand2.ToLower() == "rock") | (
hand1.ToLower() == "paper" && hand2.ToLower() == "scissors") | (hand1.
ToLower() == "rock" && hand2.ToLower() == "paper"))
{
Verdict = "You chose " + hand1 + ". The Computer chose " + hand2 + "...
Computer player wins!";
}
return Verdict;
}
//Computer random generator based on numberical value
public static string ComputerHandGenerator()
{
String ComputerHand = "";
//Random numbers 1-3
Random r = new Random();
int Random = (r.Next(1, 4));
if (Random == 1)
{
ComputerHand = "rock";
}
if (Random == 2)
{
ComputerHand = "paper";
}
if (Random == 3)
{
ComputerHand = "scissors";
}
return ComputerHand;
}
//Test 1 for Human Input
public static bool test1(string hand1)
{
return
hand1 == "rock" ||
hand1 == "scissors" ||
hand1 == "scissor" ||
hand1 == "paper";
}
//Test 2 for Computer Input
public static bool test2(string hand2)
{
return
hand2 == "rock" ||
hand2 == "scissors" ||
hand2 == "scissor" ||
hand2 == "paper";
}
}
}

**Pull request merged**